docs/designs/source-path-and-complement.md

```diff
@@ -0,0 +1,63 @@
+# Source Path
+
+In `.publish.json file`,`source_path` represents the source file's relative path to docset root.
+
+It will be used in PR comments to indicate the path of the files containing errors/warnings/suggestions.
+
+
+# Design of SourceMap
+
+For most cases, the same source_path corresponds to the same source file (i.e. One-To-One). And the source_path value is the source file's physical file path.
+
+For some special cases, it is allowed to config mappings to resolve source_path values, named **SourceMap**  
+
+|  key             |  Value            |
+|------------------|-------------------|
+| source file path | source_path value |
+
+Due to **SourceMap**, multiple source files may have the same source_path (i.e. **Many-To-One**). And the source_path value may be NOT equal to the source file's physical file path.
```

Comment:
For PowerShell, the source file is in .md but the actual input file to DocFX core is .yml. Please also take this scenario into consideration for this design.

Reply:
For powershell repo, MAML2Yaml preprocessor will be executed before building, it also leverages SourceMap.

Reply:
FYI, https://github.com/dotnet/docfx/blob/c8cb790043ae5b93173f3e28dafc28bf7f305d48/src/docfx/config/ops/OpsPreProcessor.cs#L106